File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.begin(),happiness.end());
        long long ans=0;
        int n=happiness.size()-1;
        int c=0;
        while (n>=0 && k>0){
            if(happiness[n]-c>=0)
                ans=ans+happiness[n]-c;
            else
                break;
            c++;
            n--;
            k--;
            
        }
        return ans;
        
    }
};

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness, k):
        happiness.sort()
        ans = 0
        n = len(happiness) - 1
        c = 0
        while n >= 0 and k > 0:
            if happiness[n] - c >= 0:
                ans += happiness[n] - c
            else:
                break
            c += 1
            n -= 1
            k -= 1
        return ans
